examples/cfd/stokes_mgn/pi_inference.py

```diff
@@ -0,0 +1,525 @@
+
+
+class PhysicsInformedInferencer:
+    """Class to define all the inference utils."""
+
+    def __init__(
+        self,
+        wb,
+        device,
+        gnn_u,
+        gnn_v,
+        gnn_p,
+        coords,
+        coords_inflow,
+        coords_noslip,
+        nu,
+        ref_u,
+        ref_v,
+        ref_p,
+    ):
+        super(PhysicsInformedInferencer, self).__init__()
+
+        self.wb = wb
+        self.device = device
+        self.nu = nu
+
+        self.ref_u = ref_u
+        self.ref_v = ref_v
+        self.ref_p = ref_p
+
+        self.gnn_u = torch.tensor(gnn_u).float().to(self.device)
+        self.gnn_v = torch.tensor(gnn_v).float().to(self.device)
+        self.gnn_p = torch.tensor(gnn_p).float().to(self.device)
+
+        self.coords = torch.tensor(coords, requires_grad=True).float().to(self.device)
+        self.coords_inflow = (
+            torch.tensor(coords_inflow, requires_grad=True).float().to(self.device)
+        )
+        self.coords_noslip = (
+            torch.tensor(coords_noslip, requires_grad=True).float().to(self.device)
+        )
+
+        self.model = DNN(
+            layers=[2, 128, 128, 128, 128, 3],
+            fourier_features=64,
+        ).to("cuda")
+
+        self.model = MdlsSymDNN(
+            input_keys=[Key("x"), Key("y")],
+            output_keys=[Key("u"), Key("v"), Key("p")],
+            layers=[2, 128, 128, 128, 128, 3],
+            fourier_features=64,
+        ).to(self.device)
+
+        self.node_pde = Stokes(nu=self.nu, dim=2)
+
+        self.nodes = self.node_pde.make_nodes() + [
+            self.model.make_node(name="flow_network", jit=False)
+        ]
+
+        # note: this example uses the Graph class from Modulus Sym to construct the
+        # computational graph. This allows you to leverage Modulus Sym's optimized
+        # derivative backend to compute the derivatives, along with other benefits like
+        # symbolic definition of PDEs and leveraging the PDEs from Modulus Sym's PDE
+        # module.
+        # For more details, refer: https://docs.nvidia.com/deeplearning/modulus/modulus-sym/api/modulus.sym.html#module-modulus.sym.graph
+        self.graph = Graph(
+            self.nodes,
+            [Key("x"), Key("y")],
+            [Key("u"), Key("v"), Key("p")],
+            func_arch=False,
+        ).to(
+            self.device
+        )  # For pure inference (no gradients)
+        self.graph_full = Graph(
+            self.nodes,
+            [Key("x"), Key("y")],
+            [
+                Key("u"),
+                Key("v"),
+                Key("p"),
+                Key("continuity"),
+                Key("momentum_x"),
+                Key("momentum_y"),
+            ],
+            func_arch=False,
+        ).to(self.device)
+
+        self.optimizer = torch.optim.Adam(self.model.parameters(), lr=0.001)
+
+    def parabolic_inflow(self, y, U_max=0.3):
+        u = 4 * U_max * y * (0.4 - y) / (0.4**2)
```

Comment:
Where is this 0.4 coming from? Maybe make it a variable so it has a descriptive name?

Reply:
Also, the readme says 0.41, not 0.4

Reply:
Good catch, fixed!